src/lib/plannerUtils.ts

```diff
@@ -317,6 +317,57 @@ export function isOffered(
   if (term === undefined) return true;
   return quartersOffered.find((t) => (t as Term) == term) !== undefined;
 }
+
+/**
+ * Copies a PlannerData, but changes the id's of the courses within the planner
+ * to prevent data inconsistencies
+ * Also adds a value for notes
+ * Copies labels over as well
+ * @param sourcePlanner a planner
+ * @returns a unique PlannerData instance
+ */
+export function clonePlanner(sourcePlanner: PlannerData): PlannerData {
+  const clone = { ...sourcePlanner };
+
+  const oldLabels = clone.labels;
+
+  // Create a lookup table between old ids and newStoredCourse
+  const lookup = {} as any;
+  sourcePlanner.courses.forEach((c) => {
+    lookup[c.id] = { ...c, id: uuidv4() };
+  });
+
+  clone.labels = initialLabels();
+
+  // Create a mapping between old and new label IDs
+  const labelMapping = {} as any;
+  oldLabels.forEach((oldLabel, index) => {
+    labelMapping[oldLabel.id] = clone.labels[index].id;
+  });
+
+  // Pass the new Stored courses to the clone with updated labels
+  clone.courses = Object.values(lookup).map((course: any) => ({
+    ...course,
+    labels: course.labels.map(
+      (oldLabelId: string) => labelMapping[oldLabelId] || oldLabelId,
+    ),
+  }));
+
+  // Replace all the references in the quarters to course ids with their new
+  // counterparts
+  clone.quarters = sourcePlanner.quarters.map((q) => {
+    return {
+      ...q,
+      courses: q.courses.map((crs) => {
+        return lookup[crs].id;
+      }),
+      notes: "",
```

Comment:
Remove this here and in `cloneDefaultPlanner` because quarters don't have a `notes` field